corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowserEditor.java

```diff
@@ -188,15 +194,47 @@ private void printPayload(Map.Entry<CorfuDynamicKey, CorfuDynamicRecord> entry)
             log.error("payload is NULL");
             return;
         }
+
         try {
             builder = new StringBuilder("\nPayload:\n")
                     .append(JsonFormat.printer().print(entry.getValue().getPayload()));
-            log.info(builder.toString());
+            System.out.println(builder.toString());
         } catch (Exception e) {
             log.error("invalid payload: ", e);
         }
     }
 
+    private int printTableRegistry() {
+        for (Map.Entry<TableName, CorfuRecord<TableDescriptors, TableMetadata>> entry :
+                dynamicProtobufSerializer.getCachedRegistryTable().entrySet()) {
+            try {
+                StringBuilder builder = new StringBuilder("\nKey:\n")
+                        .append(JsonFormat.printer().print(entry.getKey()));
+                System.out.println(builder.toString());
+            } catch (Exception e) {
+                log.error("Unable to print metadata entry of this registry table key {}",  entry.getKey());
+            }
+            StringBuilder builder = new StringBuilder("\nkeyType = \"");
+            builder.append("\nkeyType = \""+entry.getValue().getPayload().getKey().getTypeUrl()+"\"");
+            builder.append("\npayloadType = \""+entry.getValue().getPayload().getValue().getTypeUrl()+"\"");
+            builder.append("\nmetadataType = \""+entry.getValue().getPayload().getMetadata().getTypeUrl()+"\"");
+            builder.append("\nProtobuf Source Files: \""+
```

Comment:
this can be a very long list.  Is it valuable if printed?

Reply:
it is the only way we can match the schema to the protobuf files it brings in. without this upgrade issues become hard to debug because from the table we can't map out which protobuf files the proto definitions belong to.